xee/helpers.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Helper functions for grid parameters."""
+import math
+
+import ee
+from pyproj import Transformer
+from rasterio.transform import Affine
+import shapely
+from shapely.ops import transform
+from shapely.geometry import box
+from typing import TypedDict, Tuple, Union
+
+
+TransformType = Tuple[float, float, float, float, float, float]
+ShapeType = Tuple[int, int]
+ScalingType = Union[float, Tuple[float, float]]
+
+
+class PixelGridParams(TypedDict):
+    crs: str
+    crs_transform: TransformType
+    shape_d2: ShapeType
+
+
+def set_scale(
+    crs_transform: TransformType,
+    scaling: ScalingType,
+  ) -> list:
+  """Update the CRS transform's scale parameters."""
+  match scaling:
+    case int(xy_scale) | float(xy_scale):
```

Comment:
Given that sometimes y scale is negative and really rarely x is negative (not sure I've ever seen), I wonder if we should only allow tuple - people have to specify x and y explicitly.

Reply:
Fixed by [d661309](https://github.com/google/Xee/pull/225/commits/d661309e2d14148f3fa63dfe56c81381ce5c9807)

Reply:
Similarly: [cf48661](https://github.com/google/Xee/pull/225/commits/cf4866176a3ad065b7efad39462b763120c31a59)